src/test/components/panels/EventOverviews.test.tsx

```diff
@@ -0,0 +1,217 @@
+import { EventOverviews } from "../../../app/components/elements/panels/EventOverviews";
+import { TestUserRole, renderTestEnvironment } from "../../utils";
+import { screen, within } from "@testing-library/react";
+import {
+  mockUser,
+  mockFutureEventOverviews,
+  mockPastEventOverviews,
+} from "../../../mocks/data";
+import { rest } from "msw";
+import { API_PATH } from "../../../app/services";
+import userEvent from "@testing-library/user-event";
+
+const headers = [
+  "Actions",
+  "Title",
+  "Date",
+  "Booking deadline",
+  "Location",
+  "Status",
+  "Number confirmed",
+  "Number waiting",
+  "Number attended",
+  "Number absent",
+  "Attendance",
+];
+
+const getEventDetails = () => {
+  const table = screen.getByRole("table");
+  const rows = screen.getAllByRole("row", table);
+  const eventRow = rows[1];
+  const tableCells = within(eventRow).getAllByRole("cell");
+  return { eventRow, tableCells };
+};
+
+const getExpectedValues = ({
+  event,
+  type,
+}: {
+  event: (typeof mockFutureEventOverviews.results)[0];
+  type: "Past" | "Future";
+}) => {
+  switch (type) {
+    case "Past":
+      return [
+        "Manage",
+        `${event.title} - ${event.subtitle}`,
+        "Tue, 12 Sept 2023, 16:30",
+        "Mon, 11 Sept 2023, 16:30",
+        event.location?.address?.town,
+        event.eventStatus,
+        `${event.numberOfConfirmedBookings} / ${event.numberOfPlaces}`,
+        event.numberOfWaitingListBookings,
+        event.numberAttended,
+        event.numberAbsent,
+        `${
+          (event.numberAttended / (event.numberAttended + event.numberAbsent)) *
+          100
+        }%`,
+      ];
+    case "Future":
+    default:
+      return [
+        "Manage",
+        `${event.title} - ${event.subtitle}`,
+        "Thu, 9 Sept 9999, 20:00",
```

Comment:
The AIs that will inevitably rule the world by 9999 are going to be mad at you for this time bomb 😝
Don't change this, this is fine, but just in case you're interested, I can think of a couple of ways to deal with future dates. One would be to make the mock a function that accepts the date. That way you can always set it to today + 1 year or something like that. The other way, which I haven't looked into, would be to use Jest's `useFakeTimers`, `useRealTimers` and `setSystemTime`, but I've never tried it. It could be worth exploring the next time you need to test dates.

Reply:
If we change the approach on this one it'll probably require several changes, as future dates are littered around - in this case I just took the date that's already on one of the test events in the content repo and used that as the date for the mock event data, as I'm *relatively* confident that either Isaac CS won't be around in 9999, or the robots will have already fixed it for me :D